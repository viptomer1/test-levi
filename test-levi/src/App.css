.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #d3d3d352;
  height: 10vmin;
  display: flex;
  flex-direction: column;
  /* align-items: center; */
  justify-content: center;
  font-size: calc(10px + 1vmin);
  font-weight: normal;
  color: darkgray;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}


/* Design for Product description but we should put it in differnet css */
img {
  vertical-align: middle;
}
/* Position the image container (needed to position the left and right arrows) */
.swipe-container2 {
  position: relative;
}
/* Hide the images by default */
.mySlides {
  display: none;
}
/* Next & previous buttons */
.prev,
.next {
  cursor: pointer;
  position: absolute;
  top: 40%;
  width: auto;
  padding: 15px;
  margin-top: 40px;
  color: white;
  background-color: lightgray;
  font-size: 15px;
  border-radius: 0 3px 3px 0;
  user-select: none;
  -webkit-user-select: none;
}
/* Position the "next button" to the right */
.next {
  right: 0;
  border-radius: 3px 0 0 3px;
}

/* On hover, add a black background color with a little bit see-through */
.prev:hover,
.next:hover {
  background-color: rgba(0, 0, 0, 0.8);
}
