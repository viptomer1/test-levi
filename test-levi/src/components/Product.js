import React, { Component } from 'react';
import './home.css';
//import MOCK_DATA from '../test/mockData.js';
import constants from '../res/constants.js';
import productMockData from '../test/product.json';

class Product extends Component {
  constructor(props) {
    super(props);
    this.state = {
      product: {},
      slideIndex :1,
      selectedId : props.location.state.selectedId ? props.location.state.selectedId :'00000'
    };
  }
  
  componentDidMount() {
    fetch( constants.URL_PRODUCT_DESC +this.state.selectedId)
      .then(res => res.json())
      .then((data) => { console.log('Response product---> ',data);
        this.setState({ product: data })
      })
      .catch(console.log)
       //mock data 
      //this.setState({ product: productMockData})
  
  }
 
  // Mehtod render
  render() {  
    let product= this.state.product;
    let slideIndex = 1;
    //Method for show sliding image
    function showSlides(n) {
      var i;
      var slides = document.getElementsByClassName("mySlides");
      var dots = document.getElementsByClassName("demo");
      if (n > slides.length) {slideIndex = 1}
      if (n < 1) {slideIndex  = slides.length}
      for (i = 0; i < slides.length; i++) {
          slides[i].style.display = "none";
      }
      for (i = 0; i < dots.length; i++) {
          dots[i].className = dots[i].className.replace(" active", "");
      }
      if(slides[slideIndex -1]){
        slides[slideIndex -1].style.display = "block";
      }

    }
    //Method for sliding image right/left
    function moveSlides(n) { console.log('<<<<<<   back ./next >>>> ---> ', n);
      showSlides(slideIndex  += n);
    }
    if(product.media){
      showSlides(1 );
    }

    //rending the component
    return (
        
      <div>

      <div className="App-header" style={{ flex: 2, alignItems: 'left' }}>              
        <div className="mdl-layout__header-row" style={{ paddingLeft: '2%'}}>
            <button className="mdl-button" style={{borderWidth:'0px', float:'left'}} onClick={() => this.props.history.goBack()}>
              <i className="fa fa-angle-left" style={{fontSize:'30px',color:'grey'}}></i>
              </button>
              <span className="mdl-layout-title" style={{width:'95%',textAlign:'center',  marginLeft: '2%', display:'block'}}>Product- {product.title}</span>
        </div>
      </div>

        <div className="container">
          <div className="ca">
            {product.title ? (
              <div className="column is-half">
                
                <div className="cardsP">
                <div className="card swipe-container2" style={{
                            width: `95%`, height: `500px`,
                            //backgroundColor: 'grey'
                          }}
                        >
                    {product.media.images.urls.map((url, index) => ( 
                        <div className="mySlides"  style={{ display:  index === 0 ? 'block':'none' }}>
                          <img alt="" src={url} style={{width:"90%", height:"500px"}}/>
                        </div>
                        )) }
                      {/* <a className="prev" onClick={() =>plusSlides(-1)}>prev</a>
                      <a className="next" onClick={() =>plusSlides(1)}>next</a> */}
                      
                      <button className="prev mdl-button" style={{borderWidth:'0px', float:'left'}} onClick={() => moveSlides(-1)}>
                        <i className="fa fa-angle-left" style={{fontSize:'30px',color:'grey'}}></i>
                      </button>
                      <button className="next mdl-button" style={{borderWidth:'0px', float:'left'}} onClick={() => moveSlides(1)}>
                        <i className="fa fa-angle-right" style={{fontSize:'30px',color:'grey'}}></i>
                      </button>

                </div>
                  <div className="card media-content">
                    
                    {/* <div>{product.title}</div> */}
                    <b style={{ textTransform: "capitalize" }}>
                      {product.name}{" "}
                      <span className="tag is-primary">Â£{product.price.now}</span>
                    </b>
                    
                    <div className="is-c" style={{color:'red'}}> {product.displaySpecialOffer}.</div>
                    <div className="is-c"> {product.additionalServices.includedServices}</div>
                    
                  </div>
                </div>

                <br/>
                <div className="it-c" style={{fontSize:'20px'}}> Product Information:</div>
                <br />
                    <div className="it" dangerouslySetInnerHTML={{ __html: product.details.productInformation}} ></div>
                    <br/><div className="it">Product code: {product.code} </div>
                    <br/><hr />

                <div className="it-c" style={{fontSize:'20px'}}> Product specification:</div>
                <br /><br/>
              
                <div className="mdl-layout__header-row" style={{ paddingLeft: '2%'}}>
                    <span className="mdl-button" style={{borderWidth:'0px', float:'left'}} >Standing</span>
                      <span className="mdl-layout-title" style={{width:'95%',textAlign:'right',  marginLeft: '2%', display:'block'}}>Product- yes</span>
                </div>
                <br/><br/>    
                  
            </div>
              

            ) : (
              <div className="column">
                <span className="title has-text-grey-light">
                  No product found!
                </span>
              </div>
            )}
          </div>
        </div>
      
          </div>
      );
  }
}

export default Product;