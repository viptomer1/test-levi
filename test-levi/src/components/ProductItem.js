import React from "react";
import { useHistory } from "react-router-dom";

const ProductItem = props => { //console.log('Item>>>>>>',props);
  const { product } = props;
  const { context } = props;
  const history = useHistory();

  //TODO Method for selected id 
  function selectProduct(selId) {
    context.selectProductId = selId;
    history.push("/product" , context);
  }
  return (
    <div className="card column is-half">
      <div className="box">
        <div className="media">
          <div className="media-left">
              <img className="prodItemsImg"
                src={product.image}
                alt={product.title}
                onClick={() =>
                    context.history.push({
                        pathname: '/product',
                        state: {selectedId: product.productId}
                    })}
              />
          </div>

          <div className="media-content">
            
            <div>{product.title}</div>
            <b style={{ textTransform: "capitalize" }}>
              {product.name}{" "}
              <span className="tag is-primary">Â£{product.price.now}</span>
            </b>
            
          </div>

        </div>
      </div>
    </div>
  );
};

export default ProductItem;