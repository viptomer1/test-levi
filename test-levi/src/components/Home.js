import React, { Component } from 'react';
import ProductItem from "./ProductItem";
import './home.css';
import constants from '../res/constants.js';

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      products: [], currentProducts: [], currentPage: null ,slideIndex:0, totalPages: null, pageLimit : constants.PAGE_LIMIT
    };
  }
  
  componentDidMount() {
    fetch( constants.URL_PRODUCT_LIST )
    .then(res => res.json())
    .then((data) => { console.log('Res ---> ',data);
      this.setState({ products: data.products, totalPages: Math.ceil(data.products.length/ this.state.pageLimit), 
        currentPage:1})
    })
    .catch( console.log)
    //mock data 
    this.setState({ products: constants.allMockProducts, totalPages: Math.ceil(constants.allMockProducts.length/ this.state.pageLimit), currentPage:1})
  }

  //method for change page
  onPageChanged(n) {
    const pageLimit = this.state.pageLimit;
    let slideIndex= this.state.slideIndex;
    if (n > this.state.totalPages) {slideIndex = 1}
    if (n < 1) {slideIndex  = this.state.totalPages}
    const offset = (slideIndex - 1) * pageLimit;
    const currentProducts = this.state.products.slice(offset, offset + pageLimit);  
    console.log('<<<<<<  Home page number >>>> ---> ', slideIndex );

    let pageItems = currentProducts.map((product, index) => (
      <ProductItem
        context = {this.props}
        product={product}
        key={index}
      />
    ));
    this.setState({ pageItems, slideIndex});
  }

  //Method for sliding pages right/left
  movePages(n) { 
    this.onPageChanged(this.state.slideIndex  += n);
  }
  //Method to render first time element for 0th index
  renderFirst(){
    let slideIndex= this.state.slideIndex;
    slideIndex = slideIndex+1;
    let currentProducts = this.state.products.slice(0,  this.state.pageLimit);  
    console.log('<<<<<<  Home page first==== >>>> ---> ', slideIndex );
    let pageItems = currentProducts.map((product, index) => (
        <ProductItem
          context = {this.props}
          product={product}
          key={index}
        />
      ));
    this.setState({ pageItems, slideIndex});
  }

  render() {
    //let products = this.state.products;
    const { products, pageItems,  currentPage, totalPages ,slideIndex} = this.state;
    //let products = constants.allMockProducts;

    //rendering
    return (
      
    <div>

          
    <div className="App-header" style={{ flex: 1, alignItems: 'center' }}
      >Dishwashers ({products.length})</div>
      <br />

      <div className="container">
        { currentPage && (
                <span className="current-page d-inline-block h-100 pl-4 text-secondary">
                  Page <span className="font-weight-bold">{ slideIndex }</span> / <span className="font-weight-bold">{ totalPages }</span>
                </span>
              ) }
        <div className="cards">
       
          {products && products.length ? (
            pageItems ? pageItems: this.renderFirst()
          ) : (
            <div className="column">
              <span className="title has-text-grey-light">
                No products found!
              </span>
            </div>
          )}
        </div>

        <button className="prev1 mdl-button" style={{borderWidth:'0px', float:'left'}} onClick={() => this.movePages(-1)}>
          <i className="fa fa-angle-left" style={{fontSize:'30px',color:'grey'}}></i>
        </button>
        <button className="next1 mdl-button" style={{borderWidth:'0px', float:'right'}} onClick={() => this.movePages(1)}>
          <i className="fa fa-angle-right" style={{fontSize:'30px',color:'grey'}}></i>
        </button>

      </div>
    
      </div>
    );
  }
}

export default Home;