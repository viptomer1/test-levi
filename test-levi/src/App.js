// import logo from './logo.svg';
import './App.css';


import React, { Component } from 'react';
import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';
import Home from './components/Home';
import Product from './components/Product';

class App extends Component { //TODO add font image for back button
  render() {
    return ( <div className="App">my adpp</div>
    // <Router>
    //     <div>
    //       {/* <h2>Welcome to React Router Tutorial</h2>
    //       <nav className="navbar navbar-expand-lg navbar-light bg-light">
    //       <ul className="navbar-nav mr-auto">
    //         <li><Link to={'/'} className="nav-link"> Home </Link></li>
    //         <li><Link to={'/product'} className="nav-link">Prod</Link></li>
    //       </ul>
    //       </nav>
    //       <hr /> */}
    //       <Switch>
    //           <Route exact path='/' component={Home} />
    //           <Route path='/product' component={Product} />
    //       </Switch>
    //     </div>
    //   </Router>
    );
  }
}

/*
function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header>
    </div>
  );
}*/

export default App;