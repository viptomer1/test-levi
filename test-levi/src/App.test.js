import { render, screen } from '@testing-library/react';
import { createRenderer } from 'react-dom/test-utils';
//import renderer from 'react-test-renderer';
import App from './App';
import Home from './components/Home';
import Product from './components/Product';
import ProductItem from './components/ProductItem';

// TESTS for components and screen wise
describe('TESTs App: =>', () => {
  const commonDataprops = {"product":{ state : {}
    }
  }

  test('renders -react app simple', () => {
    render(<App />);
    const linkElement = screen.getByText(/Dishwashers/i);
    expect(linkElement).toBeInTheDocument();
  });

  test('renders Home product screen', () => {
    render(<Home />);
    let productItem = screen.getByText(/Dishwashers/i);
    expect(productItem).toBeInTheDocument();
  });


  test('Test: renders Product item', () => {
    const props = {"product":{
      title: "washer1",
      image: "//johnlewis.scene7.com/is/image/JohnLewis/238008957?",
      "price":{"now":"120"
      }
    }};

    let item=   render(<ProductItem {...props}/>);
    //console.log('DIrect >>>--> ', item);
    let productItem = screen.getByText(/washer1/i);
    expect(productItem).toBeInTheDocument();
  });

  //Test for no items found
  test('renders Product No item found', () => {
    const props = {"product":{
      },
      location:{ state : { product:{}}

      }
    };

    render(<Product {...props}/>);
    let product = screen.getByText(/Product/i);
    expect(product).toBeInTheDocument();
  });

  //Test for no product list found
  test('renders Home No item found', () => {
    const props = {"product":[]
    };

    render(<Home {...props}/>);
    let product = screen.getByText(/Dishwashers \(0/i);
    expect(product).toBeInTheDocument();
  });

  // test('check product item displayed- >', () => {  
  //   const props = {
  //         title: "item",
  //         image: "//johnlewis.scene7.com/is/image/JohnLewis/238008957?"
  //       };
  //   //let item = createRenderer.create(<ProductItem {...props} />);
  //   //SpinnerComponent = mount(<Spinner {...props} />);
  //     //expect(SpinnerComponent.find('p').text()).toBeString();
  //   const item = renderer.create(<ProductItem {...props} />).toJSON();
  //   expect(item.hasClass('prodItemsImg')).toEqual(true);
  // });

  // test('Loaded Diswasher Home product list', () => {
  //   render(<App />);
  //   const linkElement = screen.getByText(/Dishwashers/i);
  //   expect(linkElement).toBeInTheDocument();
  // });

  // test('Loaded Diswasher description', () => {
  //   render(<App />);
  //   const linkElement = screen.getByText(/Dishwashers/i);
  //   expect(linkElement).toBeInTheDocument();
  // });
});
