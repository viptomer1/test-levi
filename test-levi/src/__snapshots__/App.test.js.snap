// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly react-test-renderer 1`] = `
<div
  className="card column is-half"
>
  <div
    className="box"
  >
    <div
      className="media"
    >
      <div
        className="media-left"
      >
        <img
          alt="item"
          className="prodItemsImg"
          onClick={[Function]}
          src="//johnlewis.scene7.com/is/image/JohnLewis/238008957?"
        />
      </div>
      <div
        className="media-content"
      >
        <div>
          item
        </div>
        <b
          style={
            Object {
              "textTransform": "capitalize",
            }
          }
        >
           
          <span
            className="tag is-primary"
          >
            Â£
            120
          </span>
        </b>
      </div>
    </div>
  </div>
</div>
`;
